drivers/bbram/bbram_xec.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021 Microchip Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_xec_bbram
+
+#include <drivers/bbram.h>
+#include <errno.h>
+#include <soc.h>
+#include <sys/util.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(bbram, CONFIG_BBRAM_LOG_LEVEL);
+
+#define XEC_VBAT_REGS_ADDR \
+	(DT_REG_ADDR_BY_NAME(DT_INST(0, microchip_xec_pcr), vbatr))
```

Comment:
driver seems multi-instance, why `0` here?

Reply:
The vbr_base is not instance specific to the bbram, and hence we will remove it from the bbram_xec_config structure and in bbram_xec_check_invalid(), we will refer to the registers as

`struct vbatr_regs *const regs = (const struct vbatr_regs *)(DT_REG_ADDR_BY_NAME(DT_INST(0, microchip_xec_pcr), vbatr))`

Reply:
Please confirm and I can update accordingly.

Reply:
If some registers are common for all devices, then it makes sense, yes.

Reply:
Fixed in latest commit.